alf/environments/adv_fetch_envs/adv_fetch_env.py

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) 2022 Horizon Robotics and ALF Contributors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+
+from gym.envs.robotics.fetch_env import FetchEnv
+from gym.envs.robotics import robot_env, rotations, utils
+
+
+class AdvFetchEnv(FetchEnv):
+    def __init__(
+            self,
+            model_path,
+            n_substeps,
+            gripper_extra_height,
+            block_gripper,
+            has_object,
+            target_in_the_air,
+            target_offset,
+            obj_range,
+            target_range,
+            distance_threshold,
+            initial_qpos,
+            reward_type,
+    ):
+        """Initializes a new Fetch environment. Almost the same with ``FetchEnv``.
+        # The only change is from 4 to 7 for ``n_actions`` where the extra 3 dims
+        represent 'xyz' euler angles rotated compared to the previous step.
+
+        .. note::
+
+            After adding the rotation angles, the robot gripper might more easily
+            have collisions with the table or itself. Maybe in the future we should
+            have the environment return collision info.
+
+            You can overwrite the function `_sample_goal()` to define new goals
+            given these extra rotation actions, also `_get_obs()` to redefine the
+            achieved goal.
+
+        Args:
+            model_path (string): path to the environments XML file
+            n_substeps (int): number of substeps the simulation runs on every call to step
+            gripper_extra_height (float): additional height above the table when positioning the gripper
+            block_gripper (boolean): whether or not the gripper is blocked (i.e. not movable) or not
+            has_object (boolean): whether or not the environment has an object
+            target_in_the_air (boolean): whether or not the target should be in the air above the table or on the table surface
+            target_offset (float or array with 3 elements): offset of the target
```

Comment:
Not clear the meaning of this argument. What does a float mean? And what does an array mean?

Reply:
> Not clear the meaning of this argument. What does a float mean? And what does an array mean?

This class init function is basically copied from the OpenAI FetchEnv. I've added comments.